samcli/lib/providers/cfn_base_api_provider.py

```diff
@@ -81,16 +82,16 @@ def extract_swagger_route(
         swagger = reader.read()
         parser = SwaggerParser(stack_path, swagger)
 
-        authorizers = parser.get_authorizers(event_type)
-        default_authorizer = parser.get_default_authorizer(event_type)
+        if not disable_authorizer:
+            authorizers = parser.get_authorizers(event_type)
+            default_authorizer = parser.get_default_authorizer(event_type)
+            LOG.debug("Found '%s' authorizers in resource '%s'", len(authorizers), logical_id)
+            collector.add_authorizers(logical_id, authorizers)
 
         routes = parser.get_routes(event_type)
-
         LOG.debug("Found '%s' APIs in resource '%s'", len(routes), logical_id)
-        LOG.debug("Found '%s' authorizers in resource '%s'", len(authorizers), logical_id)
 
         collector.add_routes(logical_id, routes)
-        collector.add_authorizers(logical_id, authorizers)
 
         if default_authorizer:
             collector.set_default_authorizer(logical_id, default_authorizer)
```

Comment:
`default_authorizer` is undefined here if the above `if not disable_authorizer` statement evaluates to `False`.

Reply:
Great catch!